### ISSUE TITLE
mechanism for a homeserver to suggest a default jitsi server to use for clients

### ISSUE THREAD

**Issue by richvdh** [matrix-org/matrix-doc#2488]
As a homeserver admin, I might like to suggest a default jitsi server other than `jitsi.riot.im` to people who connect with riot to my server.

**Comment by richvdh** [comment]
It's unclear how this would interact with the `jitsi.preferredDomain` riot config option.

It's also somewhat unclear whether this belongs in the matrix protocol at all, given jitsi integration is a riot-specific extension (so perhaps what we need is a way for servers to expose client-specific configuration parameters in a generic way?).

**Comment by turt2live** [comment]
Jitsi is in the spec under the widgets banner, but indeed there's no definition for how clients acquire their defaults.

**Comment by uhoreg** [comment]
Could possibly use https://github.com/matrix-org/matrix-doc/pull/2301 or https://github.com/matrix-org/matrix-doc/pull/2063

**Comment by olmari** [comment]
As end user perspective: The unique thing with Jitsi within Matrix (or Riot/client) is that instead of "just" a widget, there exist dedicated call button for it, making it in some sense part of at least Riot, if not Matrix.

As developer'ish perspective: Most proper 'author' to define what Jitsi ought to be used would be room admins, which would bring us back into integration manager(s) and such. Does current way of using IM also apply to riot call button method? This method would ofcourse need the IM alltogether and generally focus to make custom one ever so easier to deploy.

Next best thing would be the client well-known, that would at least point HS users towards "HS-trusted" Jitsi, while that method loses control over room specific one.

Maybe even user-overridable setting, while I don't know feasibility of that..

And ofcourse all of these could be supported at once, within well thought priorities.

And just as I was about to finish, indeed somthing like @uhoreg just linked could also work... Thing is that most privacy-minded persons will have something to say no matter what is done, at the same time this shouldn't be too hard, especially in end-user perspective.. While the issue there remains that "who should control what Jitsi-instance is used".

**Comment by turt2live** [comment]
So far the proposal is for clients to just do it through .well-known, as they're allowed to: https://github.com/matrix-org/matrix-react-sdk/pull/4348

**Comment by olmari** [comment]
I like the idea of user default Jitsi offer in .well-known, if room wants to offer somwthing else, I guess they still can(?)

**Comment by neumantm** [comment]
For anyone else stumbling across this issue:
According to the documentation added in https://github.com/vector-im/element-web/pull/13047
Element already supports using the client .well-known for the default jitsi server

**Comment by t3chguy** [comment]
Indeed it does. 

See second half of https://github.com/vector-im/element-web/blob/develop/docs/jitsi.md#configuring-element-to-use-your-self-hosted-jitsi-server

**Comment by olmari** [comment]
Yep, it came more or less after this ticket, so I guess closing as implemented should be done :)

**Comment by t3chguy** [comment]
Would probably be good for it to documented in the spec rather than a client's doc

**Comment by assodefis** [comment]
I think I did add the good line on nginx
return 200 '{"m.homeserver": {"base_url": "https://matrix.defis.info/"},"im.vector.riot.jitsi": {"preferredDomain": "ourjitsi.defis.info"}}';
but when I want to add a jitsi widget, it keep offering me a jitsi.riot.im room

![image](https://user-images.githubusercontent.com/52737326/111366832-54bebc00-8694-11eb-8f7b-fa1735247e36.png)

instead of a ourjitsi.defis.info

Should I write "preferredDomain": "https://ourjitsi.defis.info/" instead of "preferredDomain": "ourjitsi.defis.info" ?

**Comment by t3chguy** [comment]
That config only affects the call button. When using an integrations manager like your screenshot shows it always uses the one it has configured internally.

**Comment by assodefis** [comment]
OK, got it. Thank you for pointing this out.
Still, it's not working.
If I click on video call button in a conference, It creates a jitsi conference to https://app.element.io/jitsi.html#xxxxxxyyyyyy and not to ourjitsi.defis.info
So I probably write something wrong but what?

**Comment by t3chguy** [comment]
That jitsi.html is just a wrapper. After clicking join conference see whether the iframe inside is pointing.

**Comment by assodefis** [comment]
https://jitsi.riot.im/JitsiMsrfdhjwrugogrlgvaloyubg
